Moq.AutoMock/AutoMocker.cs

```diff
@@ -1018,43 +1015,37 @@ bool TryCreateArguments(ConstructorInfo constructor, ObjectGraphContext context,
                 {
                     return false;
                 }
+
+                TryCache(parameters[i].ParameterType, service);
                 arguments[i] = service;
             }
             return true;
         }
+
+    }
+
+    private void TryCache(Type type, IInstance instance)

```

Comment:
not: Not sure I like the name TryCache on a method that doesn't return a book. Perhaps more like CacheIfNotFound?